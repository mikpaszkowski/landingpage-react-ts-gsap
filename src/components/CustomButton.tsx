import * as React from "react";
import styled from "styled-components";
import { colors, fontTypes } from "../styles/styleConstants";

const Button = styled.button`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  border: none;
  border-radius: 8px;
  background-color: ${(props: ButtonProps) =>
    props.color ? props.color : colors.primaryTheme};

  & > p {
    font-family: ${fontTypes.secondaryFont};
    font-size: ${(props: ButtonProps) =>
      props.fontsize ? props.fontsize : "2.5rem"};
    font-style: normal;
    font-weight: normal;
    line-height: ${(props: ButtonProps) =>
      props.lineHeight ? props.lineHeight : "3.6rem"};
  }
`;

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  color?: string;
  children?: React.ReactNode;
  lineHeight: string;
  fontsize: string;
}

export const CustomButton = ({ children, ...props }: ButtonProps) => (
  <Button {...props}>
    <p>{children}</p>
  </Button>
);
