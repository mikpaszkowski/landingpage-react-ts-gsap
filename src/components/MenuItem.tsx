import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../styles/styleConstants";

const MenuItemWrapper = styled.div`
  margin: 0.7rem 0;
`;

const MenuItemText = styled.p`
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
`;

interface MenuItemProps {
  name: string;
}

export const MenuItem = ({ name }: MenuItemProps) => {
  return (
    <MenuItemWrapper>
      <MenuItemText>{name}</MenuItemText>
    </MenuItemWrapper>
  );
};
