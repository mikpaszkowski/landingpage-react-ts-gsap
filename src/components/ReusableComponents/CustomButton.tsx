import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/styleConstants";

const Button = styled.button`
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: ${fontTypes.secondaryFont};
  font-size: ${(props: ButtonProps) =>
    props.fontSize ? props.fontSize : "2.5rem"};
  font-style: normal;
  font-weight: normal;
  line-height: ${(props: ButtonProps) =>
    props.lineHeight ? props.lineHeight : "3.6rem"};
  color: ${({ theme }) => theme.palette.common.black};
  border: none;
  border-radius: ${(props: ButtonProps) =>
    props.borderRadius ? props.borderRadius : "8px"};
  background-color: ${(props: ButtonProps) =>
    props.color ? props.color : ({ theme }) => theme.palette.common.theme};
  cursor: pointer;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;

  &::after {
    content: "";
    position: absolute;
    z-index: -1;
    bottom: -100%;
    left: 0;
    height: 100%;
    width: 100%;
    border-radius: ${(props: ButtonProps) =>
      props.borderRadius ? props.borderRadius : "8px"};
    background-color: black;
    color: ${({ theme }) => theme.palette.common.theme};
    transition: all 0.15s cubic-bezier(0.455, 0.03, 0.515, 0.955);
  }

  &:hover {
    box-shadow: 0px 1rem 2rem rgba(0, 0, 0, 0.25);
    color: ${({ theme }) => theme.palette.common.theme};

    &::after {
      transform: translateY(-100%);
    }
  }
`;

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  color?: string;
  name?: string;
  lineHeight?: string;
  fontSize?: string;
  borderRadius?: string;
}

export const CustomButton = ({ name, ...props }: ButtonProps) => (
  <Button {...props}>{name}</Button>
);
