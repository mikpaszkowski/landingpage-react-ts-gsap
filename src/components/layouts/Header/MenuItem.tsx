import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/styleConstants";
import { ReactComponent as FocusIcon } from "./ellipse-4.svg";

const MenuItemWrapper = styled.div`
  margin: 0.7rem 0;
`;

const MenuItemText = styled.a`
  position: relative;
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  text-decoration: none;
  color: ${({ theme }) => theme.palette.primary.font};

  &:focus > svg {
    display: block;
  }
`;

const FocusSVG = styled(FocusIcon)`
  display: none;
  position: absolute;
  bottom: -0.8rem;
  left: 50%;
  transform: translateX(-50%);
`;

interface MenuItemProps {
  name: string;
}

export const MenuItem = ({ name }: MenuItemProps) => {
  return (
    <MenuItemWrapper>
      <MenuItemText href="#">
        {name}
        <FocusSVG />
      </MenuItemText>
    </MenuItemWrapper>
  );
};
