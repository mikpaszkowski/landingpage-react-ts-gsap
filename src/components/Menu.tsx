import * as React from "react";
import styled from "styled-components";
import { MenuItem } from "./MenuItem";
import { CustomButton } from "./CustomButton";

const MenuWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 64.4rem;
`;

export const Menu = () => {
  return (
    <MenuWrapper>
      <MenuItem name="Home" />
      <MenuItem name="About" />
      <MenuItem name="Courses" />
      <MenuItem name="News" />
      <MenuItem name="Contact" />
      <CustomButton fontsize="2.4rem" lineHeight="3.6rem">
        Log in
      </CustomButton>
    </MenuWrapper>
  );
};
