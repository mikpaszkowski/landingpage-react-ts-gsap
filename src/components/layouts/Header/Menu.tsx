import * as React from "react";
import styled from "styled-components";
import { MenuItem } from "./MenuItem";
import { CustomButton } from "../../ReusableComponents/CustomButton";
import { device } from "../../../styles/responsive";

const MenuWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 64.4rem;

  @media ${device.mobileMedium} {
    display: none;
  }
`;

export const Menu = () => {
  return (
    <MenuWrapper>
      <MenuItem name="Home" />
      <MenuItem name="About" />
      <MenuItem name="Courses" />
      <MenuItem name="News" />
      <MenuItem name="Contact" />
      <CustomButton fontsize="2.4rem" lineHeight="3.6rem">
        Log in
      </CustomButton>
    </MenuWrapper>
  );
};
