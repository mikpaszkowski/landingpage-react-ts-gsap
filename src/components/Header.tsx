import * as React from "react";
import { Logo } from "./Logo";
import { Menu } from "./Menu";
import styled from "styled-components";
import { ReactComponent as Bar } from "../assets/images/svg-images/menu-bar.svg";
import { device } from "../styles/responsive";
import { MenuMobile } from "./MenuMobile";
import { colors } from "../styles/styleConstants";

const HeaderWrapper = styled.header`
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 10;
  max-width: 145rem;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  transform: translateX(-50%);
  background-color: ${colors.background};

  @media ${device.mobileMedium} {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
`;
const MenuBar = styled(Bar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;

  @media ${device.mobileMedium} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
  }
`;

const { useState } = React;

export const Header = () => {
  const [menuBarOpen, setMenuBarOpen] = useState<boolean | null>(null);

  const handleClick = () => {
    setMenuBarOpen(!menuBarOpen);
  };

  return (
    <HeaderWrapper>
      <Logo />
      <Menu />
      <MenuBar onClick={handleClick} />
      <MenuMobile isMenuOpen={menuBarOpen} closeMenuMobile={handleClick} />
    </HeaderWrapper>
  );
};
