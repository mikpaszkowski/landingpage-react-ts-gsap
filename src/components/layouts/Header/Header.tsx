import * as React from "react";
import { Logo } from "../../ReusableComponents/Logo/Logo";
import { Menu } from "./Menu";
import styled from "styled-components";
import { ReactComponent as Bar } from "./menu-bar.svg";
import { menuItems } from "../../staticContents/menu/menuItems";
import { device } from "../../../styles/responsive";
import { MenuMobile } from "./MenuMobile";

const HeaderWrapper = styled.header`
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 9999;
  max-width: 145rem;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  transform: translateX(-50%);
  background-color: ${({ theme }) => theme.palette.common.background};

  @media ${device.laptopSmall} {
    padding: 2.7rem 6rem;
  }

  @media ${device.mobileMedium} {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
`;
const MenuBar = styled(Bar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;
  fill: ${({ theme }) => theme.palette.primary.font};

  @media ${device.tabletLarge} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
    fill: ${({ theme }) => theme.palette.primary.font};
  }
`;

const { useState } = React;

export const Header = () => {
  const [menuBarOpen, setMenuBarOpen] = useState<boolean | null>(null);

  const handleClick = () => {
    setMenuBarOpen(!menuBarOpen);
  };

  return (
    <HeaderWrapper>
      <Logo />
      <Menu menuItems={menuItems} />
      <MenuBar onClick={handleClick} />
      <MenuMobile
        isMenuOpen={menuBarOpen}
        closeMenuMobile={handleClick}
        menuItems={menuItems}
      />
    </HeaderWrapper>
  );
};
