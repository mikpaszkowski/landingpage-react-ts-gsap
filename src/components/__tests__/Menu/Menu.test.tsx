import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithDarkTheme } from "../helpers/index";
import { menuItems } from "../../../assets/menuItems/menuItems";
import { Menu } from "../../layouts/Header/Menu";
import { MenuItem } from "../../layouts/Header/MenuItem";
import { CustomButton } from "../../ReusableComponents/CustomButton";

afterEach(() => {
  cleanup();
});

test("check render correctly <Menu /> component", () => {
  const tree = renderWithLightTheme(<Menu menuItems={menuItems} />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("check Menu component", () => {
  const wrapper = mountWithDarkTheme(<Menu menuItems={menuItems} />);

  test("should contain four(4) menu items", () => {
    expect(wrapper.find(MenuItem).length).toEqual(5);
  });

  describe("should contain button", () => {
    test("one CustomButton", () => {
      expect(wrapper.find(CustomButton).length).toEqual(1);
    });

    test("with a name Log in", () => {
      expect(wrapper.find(CustomButton).text()).toEqual("Log in");
    });
  });
});
