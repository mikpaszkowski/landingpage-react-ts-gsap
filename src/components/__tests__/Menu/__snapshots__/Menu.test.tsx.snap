// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <Menu /> component 1`] = `
.c1 {
  margin: 0.7rem 0;
}

.c2 {
  position: relative;
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #131E07;
}

.c2:focus > svg {
  display: block;
}

.c3 {
  display: none;
  position: absolute;
  bottom: -0.8rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: 'Reem Kufi',sans-serif;
  font-size: 2.4rem;
  font-style: normal;
  font-weight: normal;
  line-height: 3.6rem;
  color: #131E07;
  border: none;
  border-radius: 8px;
  background-color: #FABD20;
  cursor: pointer;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;
}

.c4::after {
  content: "";
  position: absolute;
  z-index: -1;
  bottom: -100%;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  background-color: black;
  color: #FABD20;
  -webkit-transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
}

.c4:hover {
  box-shadow: 0px 1rem 2rem rgba(0,0,0,0.25);
  color: #FABD20;
}

.c4:hover::after {
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 64.4rem;
}

@media (max-width:1110px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <a
      className="c2"
      href="#"
    >
      Home
      <svg
        className="c3"
      >
        ellipse-4.svg
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <a
      className="c2"
      href="#"
    >
      About
      <svg
        className="c3"
      >
        ellipse-4.svg
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <a
      className="c2"
      href="#"
    >
      Courses
      <svg
        className="c3"
      >
        ellipse-4.svg
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <a
      className="c2"
      href="#"
    >
      News
      <svg
        className="c3"
      >
        ellipse-4.svg
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <a
      className="c2"
      href="#"
    >
      Contact
      <svg
        className="c3"
      >
        ellipse-4.svg
      </svg>
    </a>
  </div>
  <button
    className="c4"
    fontSize="2.4rem"
  >
    Log in
  </button>
</div>
`;
